### ISSUE TITLE
extraCliparts declared in cewe2pdf.ini ignored because of redeclaration of clipartDict in convertMcf

### ISSUE THREAD

**Issue by wombatstampede** [bash0/cewe2pdf#123]
Hi all,
as my installation has no cliparts_default.xml I declared the little amount of cliparts I require in the "extraClipArts" section of cewe2pdf.ini. But cewe2pdf still complained that it couldn't find the file name for the clipart id.

My knowledge of python is minimal but it seems that the double initialisation of "clipartDict" in convertMcf silently declares a local instance of the variable (it is also initialized and declared in global scope) and therefore the contents assigned inside of convertMcf will be inivisible in other scopes.

It seems that just commenting out the double initialisation solved the problem. It might also be possible that adding clipartDict to the global declaration inside convertMcf does the trick.

```
def convertMcf(mcfname, keepDoublePages: bool, pageNumbers=None):
    global clipartPathList  # pylint: disable=global-statement
    global fontSubstitutions  # pylint: disable=global-statement
    global passepartoutDict  # pylint: disable=global-statement
    global passepartoutFolders  # pylint: disable=global-statement

    #clipartDict = dict()    #<== I commented out this line
    clipartPathList = tuple()
    (...)
```

**Comment by YLG80** [comment]
Hello,
Thanks a lot for sharing your solution.
Could you share your cewe2pdf.ini file. 
I have problems with the cliparts paths. (Windows 10)
It seems that all files are not in ProgramData but in Program Files.
Do we have to specify the full path or just the path to the Background or Clipart folders?
Thanks 
Yves
BTW Excellent script. Thanks for sharing it.
Cairo is a bit difficult to install though...but with a GTK install it' s OK.

**Comment by wombatstampede** [comment]
Replace <username> with your Windows Account Name. The software variant used in this case is "Pixum Fotowelt"
```
[DEFAULT]
cewe_folder = C:\Program Files\Pixum\Pixum Fotowelt

fontFamilies =
	Roboto,Roboto-Medium,Roboto-Black,Roboto-Italic,Roboto-BoldItalic
	#Bodoni,Bodoni,BodoniB,BodoniI,BodoniBI

extraBackgroundFolders =
	C:\Program Files\Pixum\Pixum Fotowelt\Resources\photofun\backgrounds
	C:\Program Files\Pixum\Pixum Fotowelt\Resources\photofun\backgrounds\201\201
	C:\Program Files\Pixum\Pixum Fotowelt\Resources\photofun\backgrounds\275\275

extraClipArts=
	13484, C:\Users\<username>\AppData\Local\CEWE\hps\1291\photofun\decorations\126837\13484-DECO-EMBOSS-SF\13484-DECO-EMBOSS-SF-mask.clp
	122151, C:\Users\<username>\AppData\Local\CEWE\hps\1291\photofun\decorations\122151\12271-CLIP-CW\12271-CLIP-CW.clp
	68411, C:\Program Files\Pixum\Pixum Fotowelt\Resources\photofun\decorations\68411\4808-CLIP-SF\4808-CLIP-SF.clp

#68411, arrows, Pfeil
#122151, travel, Kompass


```

**Comment by YLG80** [comment]
That is coool !
Thanks so much.
I'm going to give it a try.
Even the cewe_folder produces a root error .
I will double check if's not an issue with the txt file format. (UNICODE or UTF)
Regards
Yves

**Comment by YLG80** [comment]
OK thanks to your information and the change in the  cewe2pdf.py file, I've fixed my issues.
The first one was that Powershelll window switched to CMD cannot be used :(.
 For some reasons the path are no longer correct. 
We must use the direct cmd command file
I have switched the text files to PC/UTF8.

There are so many cliparts in my Photo Book that I have decided to create all clipart addresses with Excel  and I have integrated them in cewe2pdf.ini file.
Now the pdf output is really nice and complete. Only a few glitches.

**Config :** 
Windows10 x64
Python 3.12 (Cairo installed with a GTK x64 distribution with all dependecies
Cewe Photoservice software version 7.3.3 x64
File saved as mcf format.

Thanks for your help and thanks to the developer.
Yves
